### PR TITLE
Update and simplify the component template

### PR DESCRIPTION
**Pull request by Jutanium**
## Changes

- Remove the template's opinionated monorepo setup
- Use architecture from @Princesseuh's [component template](https://github.com/astro-community/component-template), but remove ESLint config to match the other templates
- Explain how to write tests in `test/README.md`. Note that `astro-component-tester` is included in this template. If [testing utils](https://github.com/withastro/astro/pull/4200) are added to core, we will update this template to use those.
- Include a template README that matches the other templates and includes `npm publish` and `npm link` as relevant commands

I also noticed that there was an unused variable in `create-astro`, so I updated the file to use it as intended.

## Docs
There's a [page on publishing packages](https://docs.astro.build/en/guides/publish-to-npm/) that predates @sarah11918. If this PR is approved, I will rewrite the page to use this template, and link to other resources for a monorepo setup.

### CODE REVIEW

File: examples/component/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (Princesseuh) [inline examples/component/tsconfig.json:1]: Could you update this to how it is in the other templates?

File: packages/create-astro/src/index.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ export async function main() {
 
 	// Don't touch the template name if a GitHub repo was provided, ex: `--template cassidoo/shopify-react-astro`
 	const isThirdParty = options.template.includes('/');
-	const templateTarget = options.template.includes('/')
+	const templateTarget = isThirdParty
```
Comment (Princesseuh) [inline packages/create-astro/src/index.ts:121]: I think this should be in another PR, just so you don't have to do a changeset unnecessarily

File: examples/component/test/example.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { expect } from 'chai';
+import { getComponentOutput } from 'astro-component-tester';
```
Comment (Princesseuh) [inline examples/component/test/example.test.js:2]: I'm not sure on having this inside this template. Both `astro-component-tester` and a test setup. I agree that it's useful - but it feels weirdly convoluted and opinionated for a general template. I think I'd remove all of it tbh

File: packages/create-astro/src/index.ts (hunk 2 above)
Comment (Jutanium) [reply packages/create-astro/src/index.ts]: Reverted. PR here: #4605

File: examples/component/test/example.test.js (hunk 3 above)
Comment (Jutanium) [reply examples/component/test/example.test.js]: Removed, thanks

File: examples/component/package.json [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+  ],
+  "keywords": [
+    "astro-component"
+  ],
+  "scripts": {},
+  "devDependencies": {
+    "@types/node": "^18.6.2",
+    "@types/prettier": "^2.6.4",
+    "astro": "^1.1.2",
+    "typescript": "^4.7.4"
```
Comment (Princesseuh) [inline examples/component/package.json:20]: Doesn't really need the two types packages and typescript. They're good additions but we don't have them in the other packages

**Comment by Princesseuh** [comment]
To fix the lint issue [you'll have to update the root `package.json` to remove the now-removed directory](https://github.com/withastro/astro/blob/main/package.json#L33-L35). This happens because Prettier tries to scan all of those workspaces and crash if one doesn't actually exists

File: examples/component/README.md [hunk 5]
```diff
@@ -1,47 +1,37 @@
...
-└── packages/
-    └── my-component/
-        ├── index.js
-        └── package.json
+├── index.ts
+├── src
+│   └── MyComponent.astro
+├── test
+│   ├── README.md
+│   └── example.test.js
```
Comment (Princesseuh) [inline examples/component/README.md:23]: Should be updated

**Review by Princesseuh — APPROVED** [review]

**Review by natemoo-re — APPROVED** [review]
